tests/bluetooth/bsim_bt/bsim_test_mesh/prj.conf

```diff
@@ -34,5 +34,11 @@ CONFIG_BT_MESH_FRIEND_LPN_COUNT=5
 CONFIG_BT_MESH_APP_KEY_COUNT=2
 CONFIG_BT_MESH_MODEL_KEY_COUNT=2
 CONFIG_BT_MESH_IV_UPDATE_TEST=y
+CONFIG_BT_MESH_PROV=y
+CONFIG_BT_MESH_PB_ADV=y
+CONFIG_BT_MESH_PROVISIONER=y
+CONFIG_BT_MESH_PROV_DEVICE=y
+CONFIG_BT_MESH_CDB=y
+CONFIG_BT_MESH_CDB_NODE_COUNT=3
```

Comment:
Will the provisionee actually work with CDB enabled? I believe the bt_mesh_provision call will fail because a CDB isn't created. With that being said, it looks like the CDB isn't created at all yet, so none of the roles work at the moment, it seems.

This can either be solved by setting up two different configuration files, or by allowing nodes to be provisioned without creating a CDB if it is enabled, so we can use the same binary for both roles - both for this test and in general.

Reply:
@LingaoM I think this is the reason why all the other tests have started failing. I think you might need a separate prj.conf file for the provisioner (such as `prj_provisioner.conf`), that adds these entries. You can add this device to the provisioning test shell script like this:
```sh
# Account for provisioner in phy:
extra_devs=1

# Start provisioner:
Execute ./bs_${BOARD}_tests_bluetooth_bsim_bt_bsim_test_mesh_prj_provisioner_conf \
      -v=2 -s=$1 -d=1 -RealEncryption=1 \
      -testid=mesh_provision_pb_adv_on_oob

# Start device and phy:
RunTest mesh_provision_pb_adv_on_oob \
	prov_device_pb_adv_no_oob 
```
You also need to add compilation for this provisioner in tests/bluetooth/bsim_bt/compile.sh:
```sh
app=tests/bluetooth/bsim_bt/bsim_test_mesh conf_file=prj_provisioner.conf \
  compile
```

The mesh test script could use some refactoring to support this better, but I can take that separately.